cdap-data-fabric/src/test/java/co/cask/cdap/data/format/CSVStringsRecordFormatTest.java

```diff
@@ -0,0 +1,208 @@
+    Schema mapSchema = Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING));
+    Schema schema = Schema.recordOf("event", Schema.Field.of("f1", Schema.arrayOf(mapSchema)));
+    FormatSpecification formatSpec =
+      new FormatSpecification(DelimitedStringsRecordFormat.class.getCanonicalName(),
+                              schema, Collections.<String, String>emptyMap());
+
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    format.initialize(formatSpec);
+  }
+
+
+  @Test(expected = UnsupportedTypeException.class)
+  public void testMapFieldInvalid() throws UnsupportedTypeException {
+    Schema mapSchema = Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING));
+    Schema schema = Schema.recordOf("event", Schema.Field.of("f1", mapSchema));
+    FormatSpecification formatSpec =
+      new FormatSpecification(DelimitedStringsRecordFormat.class.getCanonicalName(),
+                              schema, Collections.<String, String>emptyMap());
+
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    format.initialize(formatSpec);
+  }
+
+  @Test(expected = UnsupportedTypeException.class)
+  public void testRecordFieldInvalid() throws UnsupportedTypeException {
+    Schema recordSchema = Schema.recordOf("record", Schema.Field.of("recordField", Schema.of(Schema.Type.STRING)));
+    Schema schema = Schema.recordOf("event", Schema.Field.of("f1", recordSchema));
+    FormatSpecification formatSpec =
+      new FormatSpecification(DelimitedStringsRecordFormat.class.getCanonicalName(),
+                              schema, Collections.<String, String>emptyMap());
+
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    format.initialize(formatSpec);
+  }
+
+  @Test
+  public void testStringArrayFormat() throws UnsupportedTypeException, UnexpectedFormatException {
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    format.initialize(null);
+    String body = "userX,actionY,itemZ";
+    StructuredRecord output = format.read(ByteBuffer.wrap(Bytes.toBytes(body)));
+    String[] actual = output.get("body");
+    String[] expected = body.split(",");
+    Assert.assertTrue(Arrays.equals(expected, actual));
+  }
+
+  @Test
+  public void testDelimiter() throws UnsupportedTypeException, UnexpectedFormatException {
+    DelimitedStringsRecordFormat format = new CSVRecordFormat();
+    FormatSpecification spec = new FormatSpecification(DelimitedStringsRecordFormat.class.getCanonicalName(),
+                                                       null,
+                                                       ImmutableMap.of(DelimitedStringsRecordFormat.DELIMITER, " "));
+    format.initialize(spec);
+    String body = "userX actionY itemZ";
+    StructuredRecord output = format.read(ByteBuffer.wrap(Bytes.toBytes(body)));
+    String[] actual = output.get("body");
+    String[] expected = body.split(" ");
```

Comment:
It's a bit odd that CSVRecordFormat allows overriding the delimiter.  When put to practice here, it seems more like CSVRecordFormat and TSVRecordFormat should just be instances of DelimitedStringsRecordFormat with the appropriate delimiter specified.  Instead of making these classes, RecordFormats could use instances of a Supplier<DelimitedStringsRecordFormat> that set the right delimiter.

Not a huge deal, what you have here works.  It just seems a little odd when applied in this case.